android/src/main/java/com/swmansion/gesturehandler/core/Vector.kt

```diff
@@ -0,0 +1,60 @@
+package com.swmansion.gesturehandler.core
+
+import android.view.VelocityTracker
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_DOWN
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_LEFT
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_RIGHT
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_UP
+import kotlin.math.hypot
+
+class Vector(var x: Double, var y: Double) {
+    private var unitX: Double = x
+    private var unitY: Double = y
```

Comment:
I *think* those can be `vals` as well since they are now initialized in the constructor, I'm not sure though.
```suggestion
    private val unitX: Double
    private val unitY: Double
```